catalyst_voices/apps/voices/lib/widgets/heroes/section_hero.dart

```diff
@@ -103,6 +96,17 @@ class _HeroSectionState extends State<HeroSection>
       setState(() {});
     }
   }
+
+  Future<void> _disposeAndReinitializeVideoPlayer() async {
+    if (mounted) {
+      await _controller?.dispose();
+      _controller = VideoPlayerController.asset(
+        widget.asset,
+        package: widget.assetPackageName,
+      );
+      await _initalizedVideoPlayer();
+    }
+  }
```

Comment:
Theoretically this is the correct version, after each async gap there must be a check for `mounted`:

```suggestion
  Future<void> _disposeAndReinitializeVideoPlayer() async {
    if (mounted) {
      await _controller?.dispose();
   }
   // theoretically here after disposing a controller the widget might not be mounted anymore
   if (mounted) {
      _controller = VideoPlayerController.asset(
        widget.asset,
        package: widget.assetPackageName,
      );
      await _initalizedVideoPlayer();
    }
  }
```